src/mlpack/core/data/one_hot_encoding.hpp

```diff
@@ -28,8 +28,21 @@ namespace data {
  * @param labelsIn Input labels of arbitrary datatype.
  * @param output Binary matrix.
  */
-template<typename eT, typename RowType>
+template<class RowType, template <typename> class MatType, class eT>
 void OneHotEncoding(const RowType& labelsIn,
+                    MatType<eT>& output);
```

Comment:
Honestly, here I would suggest doing something like this:

```
template<typename RowType, typename MatType>
void OneHotEncoding(const RowType& labelsIn,
                                    MatType& output);
```

and if you then need `eT` later, just use `typename MatType::elem_type`.

Reply:
Done